code/modules/antagonists/changeling/powers/mutations.dm

```diff
@@ -244,30 +246,14 @@
 		chain = firer.Beam(src, icon_state = "tentacle", time = INFINITY, maxdistance = INFINITY, beam_sleep_time = 1)
 	..()
 
-/obj/item/projectile/tentacle/proc/reset_throw(mob/living/carbon/human/H)
-	if(H.in_throw_mode)
-		H.throw_mode_off() //Don't annoy the changeling if he doesn't catch the item
-
-/mob/proc/tentacle_grab(mob/living/carbon/C)
-	if(Adjacent(C))
-		var/obj/item/grab/G = C.grabbedby(src,1)
-		if(istype(G))
-			G.state = GRAB_AGGRESSIVE //Instant aggressive grab
-
 /mob/proc/tentacle_stab(mob/living/carbon/C)
 	if(Adjacent(C))
 		var/obj/item/I = r_hand
-		if(!is_sharp(I))
+		if(!I || istype(/obj/item/gun/magic/tentacle, I))
 			I = l_hand
-		if(!is_sharp(I))
+		if(!I || istype(/obj/item/gun/magic/tentacle, I))
 			return
-
-		C.visible_message("<span class='danger'>[src] impales [C] with [I]!</span>", "<span class='userdanger'>[src] impales you with [I]!</span>")
-		add_attack_logs(src, C, "[src] pulled [C] with a tentacle, attacking them with [I]") //Attack log is here so we can fetch the item they're stabbing with.
```

Comment:
Prob still want to log this happening

Reply:
Melee attack should automatically log it

Reply:
Pretending it is a regular attack, not a cling one. Not sure if this is desired behaviour though

Reply:
Done